### DISCUSSION TITLE
Enormous amount of scans in client 5.0.0

### DISCUSSION THREAD

**Discussion by tdanylchuk** [aerospike/spring-data-aerospike#536] [Team Posts]
Hello @rbotzer ,

In https://github.com/Playtika/testcontainers-spring-boot we faced with an issue of a changed amount of scans after performing a common query operation: 1 -> 6.

With the last version of aerospike client ```5.0.0``` our test https://github.com/Playtika/testcontainers-spring-boot/blob/develop/embedded-aerospike/src/test/java/com/playtika/test/aerospike/AerospikeScansTest.java#L36
failing with 6 scans performed on aerospike service after running 
```
 QueryPolicy policy = new QueryPolicy();
        Statement statement = new Statement();
        statement.setNamespace(namespace);
        client.query(policy, statement);
```
with a default policy.

I was curious, is this something normal or not expected?

BR, Taras.

**Comment by rbotzer** [comment]
I'm not sure I understand what you mean by changed amount of scans. Could you clarify the issue in the first paragraph?

I don't expect to see any new failure with the 5.0.0 Java client. Can you open a new ticket with our support system? You should have access to it by emailing support@aerospike.com - please do that with this issue.

**Comment by rbotzer** [comment]
And please add more detail about what the scan is like, does it use PredExp, what do you expect to see. Also what is the error code you get when it fails.

**Comment by tdanylchuk** [comment]
Sorry, I was not clear in my initial post, let me elaborate:

I don't have errors, it just after running a snippet of the code I included in the first message, on aerospike server we have 6 scans with the latest `5.0.0` client:
```
module=scan:trid=9983709198866076930:job-type=basic:ns=TEST:priority=2:status=done(ok):job-progress=100.00:run-time=56:time-since-done=1399:recs-read=0:net-io-bytes=30
module=scan:trid=1450402865600368839:job-type=basic:ns=TEST:priority=2:status=done(ok):job-progress=100.00:run-time=54:time-since-done=1457:recs-read=0:net-io-bytes=30
module=scan:trid=13113573259459337864:job-type=basic:ns=TEST:priority=2:status=done(ok):job-progress=100.00:run-time=54:time-since-done=1513:recs-read=0:net-io-bytes=30
module=scan:trid=18201274017722113435:job-type=basic:ns=TEST:priority=2:status=done(ok):job-progress=100.00:run-time=68:time-since-done=1568:recs-read=0:net-io-bytes=30
module=scan:trid=12726648231634474339:job-type=basic:ns=TEST:priority=2:status=done(ok):job-progress=100.00:run-time=59:time-since-done=1637:recs-read=0:net-io-bytes=30
module=scan:trid=16131615496841568393:job-type=basic:ns=TEST:priority=2:status=done(ok):job-progress=100.00:run-time=67:time-since-done=1698:recs-read=0:net-io-bytes=30
```

whereas same code but with 4.1.7 client we have:
```
module=scan:trid=16548863990512874253:job-type=basic:ns=TEST:priority=2:status=done(ok):job-progress=100.00:run-time=63:time-since-done=667:recs-read=0:net-io-bytes=30
```

The version of the aerospike server: 4.5.0.19

**Comment by rbotzer** [comment]
Very weird. Does your test cluster have 6 nodes, by chance?

Please email these details to support@aerospike.com - I gave them a heads up this is coming.

**Comment by tdanylchuk** [comment]
The test is against a single node of aerospike. I will mail details.

**Comment by tdanylchuk** [comment]
Have submitted ticket:

New Case: 00022269
Subject: Aerospike 5.0.0 client performs multiple scans on single query

**Comment by mtendjou** [comment]
Thanks. I see the case, but we (Aerospike Support) are not allowed to respond if the email address it comes from is not in our backend system... so we have to do some checks with the relevant contacts for that account before authorizing the team to work on it, etc... But will hopefully sort it out and get back to you through the case sooner rather than later.

**Comment by tdanylchuk** [comment]
thank you for your support, looking forward hearing from you!

**Comment by mtendjou** [comment]
Ok, I see the Support team is taking care of you. 

But I also think the issue may simply be caused by running incompatible versions... The release notes for the Client 5.0.0 mention:

- CLIENT-1369 Remove legacy client code for old servers since this client now requires server version 5.2.0.4+.

So, I wouldn't run the 5.0.0 client on any server prior to 5.2.0.4 and it seems you were running on server 4.5. But I'll let the Support team confirm too.

**Comment by rbotzer** [comment]
I'd like to brief you in person when we have our call, hopefully Thursday or Friday this week. 
Let's talk about the issue then in more detail.

**Comment by tdanylchuk** [comment]
I'm available both days
